### PR TITLE
Adds PassiveHealthCheck Fields to ServiceDefaults and IngressGateway

### PR DESCRIPTION
**Pull request by missylbytes**
### Description
Adds documentation for the PassiveHealthCheck field for ServiceDefaults and IngressGateway.

### Testing & Reproduction steps

- [x] TODO: @missylbytes needs to test the JSON and HCL values (case matching, etc)

**Note** will add the backport labels for 1.14 and 1.16 when the consul-k8s patch releases for 1.0.10 and 1.2.2 are done.
### PR Checklist
* [x] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 1]
```diff
@@ -867,9 +888,11 @@ The following table describes the configurations for passive health checks:
...
+  | `Interval` | Specifies the time between checks. | string | `0s` |
+  | `MaxFailures` | Specifies the number of consecutive failures allowed per check interval. If exceeded, Consul removes the host from the load balancer. | integer | `0` |
+  | `EnforcingConsecutive5xx` | Specifies a percentage that indicates how many times out of 100 that Consul ejects the host when it detects an outlier status. The outlier status is determined by consecutive errors in the 500-599 response range. | integer | `100` |
+  | `MaxEjectionPercent` | The maximum % of an upstream cluster that can be ejected due to outlier detection. Defaults to 10% but will eject at least one host regardless of the value. | integer | `10` |
+  | `BaseEjectionTime` | The base time that a host is ejected for. The real time is equal to the base time multiplied by the number of times the host has been ejected and is capped by max_ejection_time (Default 300s). Defaults to 30000ms or 30s. | string | `30s` |
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:895]:
```suggestion
  | `BaseEjectionTime` | Specifies the minimum amount of time that an ejected host must remain outside the cluster before rejoining. The real time is equal to the value of the `BaseEjectionTime` multiplied by the number of times the host has been ejected. The real time can not exceed the amount of time specified inthe `max_ejection_time` field. | string | `30s` |
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 2]
```diff
@@ -1163,9 +1186,11 @@ The following table describes the configurations for passive health checks:
...
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1193]: Just to flag these as sections that need to have consistent language.

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 3]
```diff
@@ -1399,9 +1424,11 @@ The following table describes the configurations for passive health checks:
...
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1431]: Just to flag these as sections that need to have consistent language.

File: website/content/docs/connect/config-entries/service-defaults.mdx [hunk 4]
```diff
@@ -625,7 +645,9 @@ The following table describes passive health check parameters you can configure:
...
 | `MaxFailures` | Specifies the number of consecutive failures allowed per check interval. If exceeded, Consul removes the host from the load balancer. | integer | `0` |
-| `EnforcingConsecutive5xx ` | Specifies a percentage that indicates how many times out of 100 that Consul ejects the host when it detects an outlier status. The outlier status is determined by consecutive errors in the 500-599 response range. | integer | `100` |
+| `EnforcingConsecutive5xx` | Specifies a percentage that indicates how many times out of 100 that Consul ejects the host when it detects an outlier status. The outlier status is determined by consecutive errors in the 500-599 response range. | integer | `100` |
+  | `MaxEjectionPercent` | The maximum % of an upstream cluster that can be ejected due to outlier detection. Defaults to 10% but will eject at least one host regardless of the value. | integer | `10` |
+  | `BaseEjectionTime` | The base time that a host is ejected for. The real time is equal to the base time multiplied by the number of times the host has been ejected and is capped by max_ejection_time (Default 300s). Defaults to 30000ms or 30s. | string | `30s` |
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-defaults.mdx:650]: Just to flag these as sections that need to have consistent language.

**Review by trujillo-adam — COMMENTED** [review]
These look pretty good. I added a few suggestions and questions. I didn't flag all of the instances, but we should obviously make the final language consistent across the different areas of the docs.

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 5]
```diff
@@ -632,9 +651,11 @@ The following table describes the configurations for passive health checks:
...
```
[the remark this answers is not in the document]
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:658]: So it is based on an envoy field that we don't actually expose https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#envoy-v3-api-field-config-cluster-v3-outlierdetection-max-ejection-time
but it does have that default.

[the remark this answers is not in the document]
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:658]: Should I add a link to it?

[the remark this answers is not in the document]
Comment (trujillo-adam) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:658]: Yeah, I think we should say something like "The real time can not exceed the amount of time specified in Envoy's `max_ejection_time` field. Refer to the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#envoy-v3-api-field-config-cluster-v3-outlierdetection-max-ejection-time) for details."

My preference would still be to leave out the default value and put the onus on the practitioner to understand the Envoy configurations they can't set through Consul, but I won't fight too hard on that point.

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 6]
```diff
@@ -632,9 +651,11 @@ The following table describes the configurations for passive health checks:
...
-| `enforcing_consecutive_5xx` | Specifies a percentage that indicates how many times out of 100 that Consul ejects the host when it detects an outlier status. The outlier status is determined by consecutive errors in the 500-599 response range. | Integer | Proxy's default configuration, which is `100` for Envoy |
+  | `Interval` | Specifies the time between checks. | string | `0s` |
+  | `MaxFailures` | Specifies the number of consecutive failures allowed per check interval. If exceeded, Consul removes the host from the load balancer. | integer | `0` |
+  | `EnforcingConsecutive5xx` | Specifies a percentage that indicates how many times out of 100 that Consul ejects the host when it detects an outlier status. The outlier status is determined by consecutive errors in the 500-599 response range. | integer | `100` |
+  | `MaxEjectionPercent` | The maximum % of an upstream cluster that can be ejected due to outlier detection. Defaults to 10% but will eject at least one host regardless of the value. | integer | `10` |
```
[the remark this answers is not in the document]
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:657]: So I really just took these descriptions from the code that was already implemented to update these documents. It looks like the people that implemented this just copy-pasted from the envoy docs, so I think they directly map to those fields.
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#envoy-v3-api-field-config-cluster-v3-outlierdetection-max-ejection-time 
IDK how we feel about copying directly for something like this.

**Review by trujillo-adam — APPROVED** [review]
Pre-approving so you're not blocked.

File: website/content/docs/connect/config-entries/ingress-gateway.mdx (hunk 2 above)
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:1193]: done

File: website/content/docs/connect/config-entries/ingress-gateway.mdx (hunk 3 above)
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:1431]: Done

File: website/content/docs/connect/config-entries/service-defaults.mdx (hunk 4 above)
Comment (missylbytes) [reply website/content/docs/connect/config-entries/service-defaults.mdx:650]: Done

File: website/content/docs/connect/config-entries/ingress-gateway.mdx (hunk 2 above)
Comment (missylbytes) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx]: Done

File: website/content/docs/connect/config-entries/service-defaults.mdx (hunk 4 above)
Comment (missylbytes) [reply website/content/docs/connect/config-entries/service-defaults.mdx]: Done

**Pull request merged**